ledger/ledgercore/statedelta.go

```diff
@@ -133,82 +155,298 @@ func MakeStateDelta(hdr *bookkeeping.BlockHeader, prevTimestamp int64, hint int,
 	}
 }
 
-// Get lookups AccountData by address
-func (ad *AccountDeltas) Get(addr basics.Address) (basics.AccountData, bool) {
+// MakeNewAccountDeltas creates account delta
+func MakeNewAccountDeltas(hint int) NewAccountDeltas {
+	return NewAccountDeltas{
+		accts:      make([]newBalanceRecord, 0, hint*2),
+		acctsCache: make(map[basics.Address]int, hint*2),
+
+		appParams:      make(map[AccountApp]*basics.AppParams),
+		appLocalStates: make(map[AccountApp]*basics.AppLocalState),
+		assetParams:    make(map[AccountAsset]*basics.AssetParams),
+		assets:         make(map[AccountAsset]*basics.AssetHolding),
+	}
+}
+
+// GetData lookups AccountData by address
+func (ad NewAccountDeltas) GetData(addr basics.Address) (AccountData, bool) {
 	idx, ok := ad.acctsCache[addr]
 	if !ok {
-		return basics.AccountData{}, false
+		return AccountData{}, false
 	}
 	return ad.accts[idx].AccountData, true
 }
 
+// GetAppParams returns app params delta value
+func (ad NewAccountDeltas) GetAppParams(addr basics.Address, aidx basics.AppIndex) (*basics.AppParams, bool) {
+	params, ok := ad.appParams[AccountApp{addr, aidx}]
+	return params, ok
+}
+
+// GetAssetParams returns asset params delta value
+func (ad NewAccountDeltas) GetAssetParams(addr basics.Address, aidx basics.AssetIndex) (*basics.AssetParams, bool) {
+	params, ok := ad.assetParams[AccountAsset{addr, aidx}]
+	return params, ok
+}
+
+// GetAppLocalState returns app local state delta value
+func (ad NewAccountDeltas) GetAppLocalState(addr basics.Address, aidx basics.AppIndex) (*basics.AppLocalState, bool) {
+	ls, ok := ad.appLocalStates[AccountApp{addr, aidx}]
+	return ls, ok
+}
+
+// GetAssetHolding returns asset holding delta value
+func (ad NewAccountDeltas) GetAssetHolding(addr basics.Address, aidx basics.AssetIndex) (*basics.AssetHolding, bool) {
+	holding, ok := ad.assets[AccountAsset{addr, aidx}]
+	return holding, ok
+}
+
 // ModifiedAccounts returns list of addresses of modified accounts
-func (ad *AccountDeltas) ModifiedAccounts() []basics.Address {
+func (ad NewAccountDeltas) ModifiedAccounts() []basics.Address {
```

Comment:
it looks like you converted all the functions that used to be pointer receivers on AccountDeltas to be value receivers on NewAccountDeltas, is that intentional?

Reply:
oh, only some of them. OK

Reply:
I changed pointer to value for read-only functions